src/faker.ts

```diff
@@ -149,13 +168,35 @@ export class Faker {
     });
   }
 
-  seed(seed?: number | number[]): void {
-    this.seedValue = seed;
+  /**
+   * Sets the seed or generates a new one.
+   *
+   * Please note that generated values are dependent on both the seed and the number of calls that have been made since it was set.
+   *
+   * @param seed The seed to use. Defaults to a random number.
+   * @returns The seed that was set.
+   */
+  seed(seed?: number): number;
+  /**
+   * Sets the seed or generates a new one.
+   *
+   * Please note that generated values are dependent on both the seed and the number of calls that have been made since it was set.
```

Comment:
I'm not sure whether we need the full block here as well.

```suggestion
   * Please note that generated values are dependent on both the seed and the
   * number of calls that have been made since it was set.
```